src/test/java/net/kaczmarzyk/e2e/converter/OffsetDateTimeE2eTest.java

```diff
@@ -237,4 +248,37 @@ public void findsByOffsetDateTimeEqualWithCustomDateFormatWithDateOnly() throws
 				.andExpect(jsonPath("$.[?(@.firstName=='Barry')]").exists())
 				.andExpect(jsonPath("$[1]").doesNotExist());
 	}
+
+	@Test
+	public void findsByOffsetDateTimeEqualDayWithDefaultOffsetDateTimeFormat() throws Exception {
+		customer("Barry", "Benson")
+				.nextSpecialOffer(OffsetDateTime.of(2020, 7, 20, 0, 0, 0, 0, ofHours(4)))
+				.build(em);
+		customer("Adam", "Flayman")
+				.nextSpecialOffer(OffsetDateTime.of(2020, 7, 18, 23, 59, 59, 999999000, ofHours(4)))
```

Comment:
records with time `23:59:59` (time right before lowerBoundary of filtering) with above 999999000ns are returned, but they shouldn't be. Maybe because of some db or filtering constraints? Don't know if this accuracy is sufficient?

Reply:
why they shouldn't be? please explain